<?xml version="1.0"?>
<launch>
    <arg name="roomba" />
    <arg name="robot_frame_id" default="$(arg roomba)/base_link" />

    <arg name="map_file" default="$(find roomba_visualizer)/map/map.yaml" />

    <group ns="$(arg roomba)" >
        <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)" />
        <include file="$(find multi_robots)/launch/launch_devices.launch" >
            <arg name="roomba" default="$(arg roomba)" />
        </include>
        <include file="$(find roomba_visualizer)/launch/amcl.launch" >
            <arg name="roomba" default="$(arg roomba)" />
        </include>

        <node pkg="dwa_planner" type="dwa_planner" name="dwa_planner">
            <rosparam command="load" file="$(find navigator)/config/dwa_param.yaml"/>
            <param name="ROBOT_FRAME" value="$(arg robot_frame_id)" />
            <remap from="/local_map" to="local_map"/>
            <remap from="/scan" to="corrected_scan"/>
            <remap from="/local_goal" to="local_goal"/>
            <remap from="/cmd_vel" to="local_path/cmd_vel"/>
            <remap from="/odom" to="roomba/corrected_odometry"/>
        </node>

        <node pkg="navigator" type="message_reviser_node" name="message_reviser" />
        <node pkg="navigator" type="waypoints_manager_node" name="waypoint_manager">
            <param name="WITH_RVIZ" value="true" />
            <param name="GOAL_THRESHOLD" value="1.2" />
        </node>
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find navigator)/rviz/$(arg roomba).rviz" >
            <remap from="/initialpose" to="initialpose" />
            <remap from="/move_base_simple/goal" to="rviz/local_goal" />
        </node>
    </group>
</launch>
